CRM/Admin/Form/Setting/Debugging.php

```diff
@@ -54,4 +55,17 @@ public function buildQuickForm() {
     parent::buildQuickForm();
   }
 
+  /**
+   * Process the form submission.
+   */
+  public function postProcess() {
+    $params = $this->controller->exportValues($this->_name);
+
+    if ($params['isProductionEnvironment'] === FALSE) {
```

Comment:
One thing I did notices is that this is happening in the form - which is not correct, it should work via the api too - take a look at how the logging setting works for how to not do it in the form layer.